code_review/20240816_vjs/R1_depthdevelopment.Rmd

```diff
@@ -0,0 +1,1550 @@
+  axis.title.x = element_text(size = 7, family ="Arial", color = c("black")),
+  axis.title.y = element_text(size = 7, family ="Arial", color = c("black")),
+  axis.line = element_line(linewidth = .2), 
+  axis.ticks = element_line(linewidth = .2))
+
+ggsave(filename = "/Volumes/Hera/Projects/corticalmyelin_development/Figures/Figure2_supplementarya/Depth_trajectories_withdata.pdf", device = "pdf", dpi = 500, width = 2.7, height = 4.65)
+```
+
+```{r}
+ggplot() +
+#depth 1
+geom_line(data = frontal.fittedvalues.alldepths$depth_1, aes(x = age, y = fitted), color = depth_colorbar[7], linewidth = 1.8) +
+#depth 2
+geom_line(data = frontal.fittedvalues.alldepths$depth_2, aes(x = age, y = fitted), color = depth_colorbar[6], linewidth = 1.8) +
+#depth 3
+geom_line(data = frontal.fittedvalues.alldepths$depth_3, aes(x = age, y = fitted), color = depth_colorbar[5], linewidth = 1.8) +
+#depth 4
+geom_line(data = frontal.fittedvalues.alldepths$depth_4, aes(x = age, y = fitted), color = depth_colorbar[4], linewidth = 1.8) +
+#depth 5
+geom_line(data = frontal.fittedvalues.alldepths$depth_5, aes(x = age, y = fitted), color = depth_colorbar[3], linewidth = 1.8) +
+#depth 6
+geom_line(data = frontal.fittedvalues.alldepths$depth_6, aes(x = age, y = fitted), color = depth_colorbar[2], linewidth = 1.8) +
+#depth 7
+geom_line(data = frontal.fittedvalues.alldepths$depth_7, aes(x = age, y = fitted), color = depth_colorbar[1], linewidth = 1.8) +
+labs(x="\nAge", y=sprintf("R1\n")) +
+theme_classic() +
+scale_y_continuous(limits = c(0.48, 0.65)) +
+theme(
+  legend.position = "none", 
+  panel.grid.major = element_blank(),
+  panel.grid.minor = element_blank(),
+  axis.text = element_text(size = 6, family = "Arial", color = c("black")),
+  axis.title.x = element_text(size = 7, family ="Arial", color = c("black")),
+  axis.title.y = element_text(size = 7, family ="Arial", color = c("black")),
+  axis.line = element_line(linewidth = .2), 
+  axis.ticks = element_line(linewidth = .2))
+
+ggsave(filename = "/Volumes/Hera/Projects/corticalmyelin_development/Figures/Figure2_supplementarya/Depth_trajectories_allfits.pdf", device = "pdf", dpi = 500, width = 2.7, height = 4.65)
+```
+
+## Robust longitudinal increases in frontal lobe R1 
+
+```{r}
+#Compute longitudinal change in frontal lobe R1 between paired longitudinal sessions, at each cortical depth
+compute_R1_longitudinalchange <- function(input.df){
+  
+  input.df$mp2rage.date <- ymd(input.df$mp2rage.date)
+  
+  #session-to-session raw longitudinal change in R1
+  R1.longitudinal.change <- input.df %>% group_by(subject_id) %>%
+    mutate(R1.change.raw = mean.R1 - lag(mean.R1)) %>% 
+    ungroup() %>% select(subject_id, session_id, age, mp2rage.session_number, mean.R1, R1.change.raw)
+  
+  #session-to-session longitudinal change in age
+  R1.longitudinal.change <- R1.longitudinal.change %>% group_by(subject_id) %>%
+    mutate(age.change = age - lag(age)) %>% 
```

Comment:
probably fine, but `lag` without first `%>% arrange(mp2rage.date)` can give head scratching bugs later on

Reply:
can also add `mutate(visit_diff=mp2rage.date - lag(mp2rage.date))` for a sanity check (and maybe eventually an additional regressor?)